From 92a3325259667359c6cb86ec967986213aa30944 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Mon, 5 Oct 2020 10:41:49 -0500
Subject: [PATCH 61/62] arm64: dts: beacon-imx8m kits: Assign regulator to
 touchscreen

The touchscreen has a VDD pin, but Linux doesn't know the regulator is
needed by the touchscreen, and when there are no consumers of the
regulator, Linux powers them down.  This is a problem when the touch
controller is still expecting power, but it as gone away.

This patch adds the corresponding vdd-supply assignement to the touch
controller node to ensure the power it needs is readily available.

Fixes: PHOENIX8-214

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/beacon-imx8mm-lvds.dtsi | 1 +
 arch/arm64/boot/dts/freescale/beacon-imx8mn-lvds.dtsi | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-lvds.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-lvds.dtsi
index cdbcd839b96c..4d98cf61cb71 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-lvds.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-lvds.dtsi
@@ -42,6 +42,7 @@
 		reg = <0x26>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_touchscreen>;
+		vdd-supply = <&reg_lcd_reset>;
 		interrupts-extended = <&gpio1 9 IRQ_TYPE_EDGE_RISING>;
 	};
 
diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mn-lvds.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mn-lvds.dtsi
index c27b95c7caee..996f1ff8dfa5 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mn-lvds.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mn-lvds.dtsi
@@ -42,6 +42,7 @@
 		reg = <0x26>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_touchscreen>;
+		vdd-supply = <&reg_lcd_reset>;
 		interrupts-extended = <&gpio1 9 IRQ_TYPE_EDGE_RISING>;
 	};
 
-- 
2.17.1

